cloudevents-sdk-warp/src/lib.rs

```diff
@@ -0,0 +1,254 @@
+                .build()
+                .unwrap();
+
+            let resp = super::from_event(input);
+
+            assert_eq!(
+                resp.headers()
+                    .get("ce-specversion")
+                    .unwrap()
+                    .to_str()
+                    .unwrap(),
+                "1.0"
+            );
+            assert_eq!(
+                resp.headers().get("ce-id").unwrap().to_str().unwrap(),
+                "0001"
+            );
+            assert_eq!(
+                resp.headers().get("ce-type").unwrap().to_str().unwrap(),
+                "example.test"
+            );
+            assert_eq!(
+                resp.headers().get("ce-source").unwrap().to_str().unwrap(),
+                "http://localhost/"
+            );
+            assert_eq!(
+                resp.headers().get("ce-someint").unwrap().to_str().unwrap(),
+                "10"
+            );
+        }
+
+        #[tokio::test]
+        async fn test_response_with_full_data() {
+            let j = json!({"hello": "world"});
+
+            let input = EventBuilderV10::new()
+                .id("0001")
+                .ty("example.test")
+                .source(Url::from_str("http://localhost").unwrap())
+                .data("application/json", j.clone())
+                .extension("someint", "10")
+                .build()
+                .unwrap();
+
+            let resp = super::from_event(input);
+
+            assert_eq!(
+                resp.headers()
+                    .get("ce-specversion")
+                    .unwrap()
+                    .to_str()
+                    .unwrap(),
+                "1.0"
+            );
+            assert_eq!(
+                resp.headers().get("ce-id").unwrap().to_str().unwrap(),
+                "0001"
+            );
+            assert_eq!(
+                resp.headers().get("ce-type").unwrap().to_str().unwrap(),
+                "example.test"
+            );
+            assert_eq!(
+                resp.headers().get("ce-source").unwrap().to_str().unwrap(),
+                "http://localhost/"
+            );
+            assert_eq!(
+                resp.headers()
+                    .get("content-type")
+                    .unwrap()
+                    .to_str()
+                    .unwrap(),
+                "application/json"
+            );
+            assert_eq!(
+                resp.headers().get("ce-someint").unwrap().to_str().unwrap(),
+                "10"
+            );
+
+            let (_, body) = resp.into_parts();
+            let body = hyper::body::to_bytes(body).await.unwrap();
+
+            assert_eq!(j.to_string().as_bytes(), body);
+        }
+    }
+}
+
+pub mod filter {
+    use crate::server_request::request_to_event;
+
+    use cloudevents::Event;
+    use warp::http::HeaderMap;
+    use warp::Filter;
+    use warp::Rejection;
+
+    #[derive(Debug)]
+    pub struct CEFilterError {
+        message: String,
```

Comment:
More than placing the `message` here, can you place the full `message::Error`?

Reply:
Ping

Reply:
why i missed this part is a fear that this will open pandora box of `Send` trait and async rust if we do it like that

Reply:
Well, that shouldn't be a problem because `message::Error` is `Sync` and `Send`

Reply:
https://docs.rs/cloudevents-sdk/0.3.0/cloudevents/message/enum.Error.html#synthetic-implementations

Reply:
do we want to rename `message` to `error` ?